### ISSUE TITLE
Error using DataMemoryLayer

### ISSUE THREAD

**Issue by andreteixeiralopes** [BVLC/caffe#2356]
Hi, I'm trying to make the caffe training in my C++ code with the training data comming from OpenCV.

My prototxt data layer is as fllows:
layers {
  name: "mnist"
  type: MEMORY_DATA
  top: "data"
  top: "label"
  memory_data_param {
   batch_size: 60
   channels: 1
   height: 28
   width: 28
  }
  include: { phase: TRAIN }
}

my code to do the optimazation:

    Caffe::SetDevice(0);
    Caffe::set_mode(Caffe::GPU);

    SolverParameter solverParamters;
    ReadProtoFromTextFileOrDie(myPath + "mnist_solver.prototxt", &solverParamters);
    SGDSolver<float> solver(solverParamters);

    shared_ptr < MemoryDataLayer<float > > train_data_layer = boost::dynamic_pointer_cast < MemoryDataLayer<float > > (solver.net().get()->layers()[0]);
    if (!train_data_layer) {
        std::cout << "The first layer is not a MemoryDataLayer!\n";
        return -1;
    }

    train_data_layer->AddMatVector(mnistTrainSamples, mnistTrainLabels);
    solver.Solve();

When I call, solver.Solve(), I get the following error: data_ MemoryDataLayer needs to be initalized by calling Reset

If someone can help me, I will be very gratefull.

Thanks a lot!

**Comment by longjon** [comment]
Please ask usage questions on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) list. Thanks!
